### PR TITLE
WIP: Perl SDK - adds devcontainer, vaas module and tests

### PR DESCRIPTION
**Pull request by sfotiadis**
WIP

### CODE REVIEW

**Comment by unglaublicherdude** [comment]
wait. this isn't WIP anymore, isn't it?

**Comment by unglaublicherdude** [comment]
ah ok. tests are broken

File: perl/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# local env files
+.env.local
+.env.*.local
+.env
```
Comment (pstadermann) [inline perl/.gitignore:4]: .env*

File: perl/src/VaaS.pm [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+sub _create_websocket_connection {
+    my $self = shift;
+    my $url = shift;
+    my $token = shift;
+
+    $self->_create_tcp_socket($url);
+
+    $self->{'ws_client'} = Protocol::WebSocket::Client->new(url => $url);
```
Comment (pstadermann) [inline perl/src/VaaS.pm:76]: $self->ws_client =

Search for "$self->{" and replace everywhere.

File: perl/src/VaaS.pm [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        my %response_hash = %$response_ref;
+
+        return $response_hash{'access_token'};
+
+    } else {
+        print "Error: ", $response->code, $response->message, "\n";
+    }
+}
+
+sub get_verdict_by_sha256 {
```
Comment (pstadermann) [inline perl/src/VaaS.pm:147]: API is different to all other SDKs.

File: perl/src/VaaS.pm [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+sub _upload {
+    my $self = shift;
+    my $buffer = shift;
+    
+    my $response = $self->{'http_client'}->put($self->{'upload_url'}, [Authorization=>$self->{'upload_token'}], $buffer);
+
+    if ($self->{'debug'} == 1){
+        if ($response->is_success) {
+            print "Upload successful.\n";
```
Comment (pstadermann) [inline perl/src/VaaS.pm:190]: This will pollute the console of the application using the SDK.

**Review by pstadermann — CHANGES_REQUESTED** [review]
My Perl is more than rusty. Just a few things I saw on a quick review.

**Comment by unglaublicherdude** [comment]
Is this still WIP? Please also add an example of how to use the sdk.

**Comment by secana** [comment]
@sfotiadis this PR seems to be stalled. What should we do with it?

**Pull request closed without merging**